main.tf

```
@@ -106,7 +105,7 @@ resource "random_pet" "cbd" {
   keepers = {
     node_role_arn  = local.ng.node_role_arn
     subnet_ids     = join(",", local.ng.subnet_ids)
-    instance_types = join(",", local.ng.instance_types)
+    instance_types = join(",", sort(local.ng.instance_types))
```

Confirmed that changing the instance types forces a new node group:
```suggestion
    instance_types = join(",", local.ng.instance_types)
```